--- src/src/tracker/Tracker.h
+++ src/src/tracker/Tracker.h
@@ -98,7 +98,7 @@
 		PanelTop_Sample,
 		PanelTop_Instrument
 	};
-	PanelRotate panelrotate = PanelRotate::PanelTop;
+	PanelRotate panelrotate = PanelTop;
 
 	// I've replaced some constants
 #ifndef __LOWRES__
--- src/src/tracker/TrackerKeyboard.cpp
+++ src/src/tracker/TrackerKeyboard.cpp
@@ -451,18 +451,18 @@
 		return;
 
 	switch( panelrotate ){
-		case PanelRotate::PanelTop:{
-			panelrotate = PanelRotate::PanelTop_Sample;
+		case PanelTop:{
+			panelrotate = PanelTop_Sample;
 			eventKeyDownBinding_InvokeSectionSamples();
 			break;
 		}
-		case PanelRotate::PanelTop_Sample:{
-			panelrotate = PanelRotate::PanelTop_Instrument;
+		case PanelTop_Sample:{
+			panelrotate = PanelTop_Instrument;
 			eventKeyDownBinding_InvokeSectionInstruments();
 			break;
 		}
-		case PanelRotate::PanelTop_Instrument:{
-			panelrotate = PanelRotate::PanelTop;
+		case PanelTop_Instrument:{
+			panelrotate = PanelTop;
 			sectionSwitcher->showBottomSection(SectionSwitcher::ActiveBottomSectionNone);
 			break;
 		}
