+++ src/configure.ac	2025-05-10 17:39:42.082991201 +0200
@@ -225,29 +225,7 @@
     [android], [RESOLV_LIBS=""],
     [haiku],   [RESOLV_LIBS="-lnetwork"],
     [win32],   [RESOLV_LIBS=""],
-               [RESOLV_LIBS="-lresolv"])
-
-if test "x$PLATFORM" != xwin32; then
-LIBS_TMP="${LIBS}"
-LIBS="${RESOLV_LIBS}"
-AC_LINK_IFELSE([AC_LANG_SOURCE([
-        #include <sys/types.h>
-        #include <netinet/in.h>
-        #include <arpa/nameser.h>
-        #include <resolv.h>
-        #include <netdb.h>
-
-        int main() {
-            (void)res_query(NULL, 0, 0, NULL, 0);
-            return 0;
-        }
-    ])],
-    [],
-    [AC_MSG_ERROR([res_query() not found with LIBS="${LIBS}"])])
-LIBS="${LIBS_TMP}"
-PC_LIBS="${RESOLV_LIBS} ${PC_LIBS}"
-fi
-
+               [RESOLV_LIBS=""])
 fi
 
 if test "x$enable_zlib" != xno; then
